### PR TITLE
Move coverage upload to separate step

### PR DESCRIPTION
**Pull request by Czaki**
This PR move coverage to separate steep nad add coverage summary information in action summary

### CODE REVIEW

**Comment by Czaki** [comment]
Ok. Windows tests are broken from a long time, but this reveals a much deeper problem as inspecting is using imports and imported packages are present in `sys.modules` after virtualenv removal.

**Comment by Czaki** [comment]
Codecov will be fixed once the new test plugin is released. 

cc @DragaDoncila

File: tests/test_all_plugins.py [hunk 1]
```diff
@@ -16,34 +13,6 @@
...
-def plugin_env():
-    try:
-        with isolated_plugin_env(PLUGIN) as env:
-            yield env
-    except CalledProcessError as e:
-        if "Failed building wheel" in str(e.output):
-            yield None
-
-
-def test_entry_points_importable(plugin_env):
```
Comment (nclack) [inline tests/test_all_plugins.py:29]: Can you give me a bit more context around this removal? Was it always xfail-ing?

Comment (Czaki) [reply tests/test_all_plugins.py:29]: there is no CI job with `TEST_PACKAGE_NAME` defined so it land in xfail branch.

File: tests/test_fetch.py [hunk 2]
```diff
@@ -26,7 +24,7 @@ def test_fetch_npe2_manifest():
 
 @pytest.mark.skip("package looks deleted from pypi")
 def test_fetch_npe1_manifest_with_writer():
-    mf = fetch_manifest("example-plugin")
+    mf = fetch_manifest("dummy-test-plugin", version="0.1.1")
```
Comment (DragaDoncila) [inline tests/test_fetch.py:27]:
```suggestion
    mf = fetch_manifest("dummy-test-plugin", version="0.1.2")
```

**Comment by DragaDoncila** [comment]
Hmm smh now this test is failing because afaict the npe2 adapter doesn't convert npe1 themes? I don't really understand what this test is testing. It's also odd because both the sample dict of dicts error and the theme error don't occur when launching the plugin in `napari` with the npe2 adapter enabled so I don't get what the difference is here

File: tests/test_fetch.py [hunk 3]
```diff
@@ -26,7 +24,7 @@ def test_fetch_npe2_manifest():
 
 @pytest.mark.skip("package looks deleted from pypi")
 def test_fetch_npe1_manifest_with_writer():
-    mf = fetch_manifest("example-plugin")
+    mf = fetch_manifest("dummy-test-plugin", version="0.1.2")
```
Comment (DragaDoncila) [inline tests/test_fetch.py:27]:
```suggestion
    mf = fetch_manifest("dummy-test-plugin", version="0.1.3")
```

**Comment by Czaki** [comment]
So it looks ready for review :)

**Review by nclack — APPROVED** [review]
lgtm! thanks!

**Pull request merged**